packages/dynamite/dynamite/lib/src/models/path_item.dart

```diff
@@ -1,97 +1,61 @@
 
-  final List<Parameter>? parameters;
+  @BuiltValueField(compare: false)
+  String? get description;
 
-  final Operation? get;
+  BuiltList<Parameter>? get parameters;
 
-  final Operation? put;
+  Operation? get get;
 
-  final Operation? post;
+  Operation? get put;
 
-  final Operation? delete;
+  Operation? get post;
 
-  final Operation? options;
+  Operation? get delete;
 
-  final Operation? head;
+  Operation? get options;
 
-  final Operation? patch;
+  Operation? get head;
 
-  final Operation? trace;
+  Operation? get patch;
 
-  Map<String, Operation> get operations => <String, Operation>{
-        if (get != null) 'get': get!,
-        if (put != null) 'put': put!,
-        if (post != null) 'post': post!,
-        if (delete != null) 'delete': delete!,
-        if (options != null) 'options': options!,
-        if (head != null) 'head': head!,
-        if (patch != null) 'patch': patch!,
-        if (trace != null) 'trace': trace!,
+  Operation? get trace;
+
+  Map<PathItemOperation, Operation> get operations => <PathItemOperation, Operation>{
+        if (get != null) PathItemOperation.get: get!,
+        if (put != null) PathItemOperation.put: put!,
+        if (post != null) PathItemOperation.post: post!,
+        if (delete != null) PathItemOperation.delete: delete!,
+        if (options != null) PathItemOperation.options: options!,
+        if (head != null) PathItemOperation.head: head!,
+        if (patch != null) PathItemOperation.patch: patch!,
+        if (trace != null) PathItemOperation.trace: trace!,
       };
+}
 
-  PathItem copyWithOperations(final Map<String, Operation> operations) => PathItem(
-        description: description,
-        parameters: parameters,
-        get: operations['get'] ?? get,
-        put: operations['put'] ?? put,
-        post: operations['post'] ?? post,
-        delete: operations['delete'] ?? delete,
-        options: operations['options'] ?? options,
-        head: operations['head'] ?? head,
-        patch: operations['patch'] ?? patch,
-        trace: operations['trace'] ?? trace,
-      );
-
-  @override
-  bool operator ==(final Object other) =>
-      other is PathItem &&
-      const ListEquality().equals(parameters, other.parameters) &&
-      get == other.get &&
-      put == other.put &&
-      post == other.post &&
-      delete == other.delete &&
-      options == other.options &&
-      head == other.head &&
-      patch == other.patch &&
-      trace == other.trace;
-
-  @override
-  int get hashCode =>
-      const ListEquality().hash(parameters) +
-      get.hashCode +
-      put.hashCode +
-      post.hashCode +
-      delete.hashCode +
-      options.hashCode +
-      head.hashCode +
-      patch.hashCode +
-      trace.hashCode;
+enum PathItemOperation {
+  get('get'),
+  put('put'),
+  post('post'),
+  delete('delete'),
+  options('options'),
+  head('head'),
+  patch('patch'),
+  trace('trace');
+
+  const PathItemOperation(this.value);
+  final String value;
```

Comment:
as the enum names are the same as their value we could just drop the extra value and call `PathItemOperation.get.name`.

If you like the current approach for saftey or codestyle reasons I'm fine with it.

Reply:
Ah I forgot that this is possible